src/test/shell/bazel/bazel_cc_code_coverage_test.sh

```diff
@@ -164,84 +157,172 @@ function tear_down() {
+function assert_gcov_coverage_srcs_a_cc() {
+    local output_file="${1}"; shift
+
+    # The expected coverage result for coverage_srcs/a.cc in gcov format.
+    local expected_gcov_result_a_cc="file:coverage_srcs/a.cc
+function:3,1,_Z1ab
+lcount:3,1
+lcount:4,1
+branch:4,taken
+branch:4,nottaken
+lcount:5,1
+lcount:7,0"
+    assert_coverage_entry_in_file "$expected_gcov_result_a_cc" "$output_file"
+}
+
+
+# Asserts if coverage result in gcov format for coverage_srcs/t.cc is included
+# in the given output file.
+#
+# - output_file    The location of the coverage output file.
+function assert_gcov_coverage_srcs_t_cc() {
+    local output_file="${1}"; shift
+
+    # The expected coverage result for coverage_srcs/t.cc in gcov format.
+    local expected_gcov_result_t_cc="file:coverage_srcs/t.cc
+function:4,1,main
+lcount:4,1
+lcount:5,1
+lcount:6,1"
+    assert_coverage_entry_in_file "$expected_gcov_result_t_cc" "$output_file"
+}
+
+function test_cc_test_coverage_lcov() {
     # Run the C++ coverage script with the environment setup accordingly.
     # This will get coverage results for coverage_srcs/a.cc and
     # coverage_srcs/t.cc.
-    run_coverage > "$TEST_log"
-
-    # Assert that the C++ coverage output is correct.
-    # The covered files are coverage_srcs/a.cc and coverage_srcs/t.cc.
-
-    # The expected total number of lines of COVERAGE_OUTPUT_FILE
-    # is 25. This number can be computed by counting the number
-    # of lines in the variables declared below $expected_result_a_cc
-    # and $expected_result_t_cc.
-    [[ $(wc -l < "$COVERAGE_OUTPUT_FILE_VAR") == 25 ]] || \
-        fail "Number of lines $nr_of_lines is different than 25"
-
-    # The expected result can be constructed manually by following the lcov
-    # documentation and manually checking what lines of code are covered when
-    # running the test.
-    # For details about the lcov format see
-    # http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
-    # The newlines are replaced with commas to facilitate comparing the
-    # expected values with the actual results.
-
-    # The expected coverage result for coverage_srcs/a.cc.
-    local expected_result_a_cc="TN:,\
-SF:coverage_srcs/a.cc,\
-FN:3,_Z1ab,\
-FNDA:1,_Z1ab,\
-FNF:1,\
-FNH:1,\
-DA:3,1,\
-DA:4,1,\
-DA:5,1,\
-DA:7,0,\
-LF:4,\
-LH:3,\
-end_of_record"
+    run_coverage lcov > "$TEST_log"
 
-    # The expected coverage result for coverage_srcs/t.cc.
-    local expected_result_t_cc="TN:,\
-SF:coverage_srcs/t.cc,\
-FN:4,main,\
-FNDA:1,main,\
-FNF:1,\
-FNH:1,\
-DA:4,1,\
-DA:5,1,\
-DA:6,1,\
-LF:3,\
-LH:3,\
-end_of_record"
+    # Location of the output file of the C++ coverage script when lcov is used.
+    local output_file="$COVERAGE_DIR_VAR/_cc_coverage.dat"
+
+    # The expected total number of lines of output file is 25. This assertion
+    # is needed to make sure no other source files are included in the output
+    # file.
+    [[ "$(wc -l < "$output_file")" == 25 ]] || \
```

Comment:
Abstract "$(wc -l < "$output_file")" as a local variable since you're using it twice.